src/components/ButtonLink/ButtonLink.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { ButtonProps, buttonStyles, iconSizeMap } from "../Button";
+import { Icon, IconName } from "../Icon";
+
+export interface ButtonLinkProps<T extends React.ElementType>
+  extends ButtonProps {
+  as?: T;
+  children?: React.ReactNode;
+  href: string;
+  iconLeft?: IconName;
+  iconRight?: IconName;
+  target?: string;
+}
+
+export function ButtonLink<T extends React.ElementType = "a">({
+  as,
+  children,
+  className,
+  href,
+  iconLeft,
+  iconRight,
+  id,
+  size,
+  target,
+  variant,
+  width,
+  ...props
```

Comment:
Same here, only a few props needed (`as` and the ones related to styling).
Other props can be passed through `...props`.

```suggestion
  as,
  className,
  size,
  variant,
  width,
  ...props
```